### PR TITLE
ES6 refactor of DelegatedModule

### PR DESCRIPTION
**Pull request by davegonzalez**
**What kind of change does this PR introduce?**
ES6 Refactor

**Did you add tests for your changes?**
No added tests, all current tests still pass.

**If relevant, link to documentation update:**
N/A

**Summary**
Updates to make the codebase a bit more consistent with ES6 standards. This is based on issue #4099 

**Does this PR introduce a breaking change?**
No breaking changes.

**Other information**
There's also this PR open https://github.com/webpack/webpack/pull/4345, though I'm not sure if it's still being worked on.

### CODE REVIEW

File: lib/DelegatedModule.js [hunk 1]
```diff
@@ -2,79 +2,83 @@
...
+class DelegatedModule extends Module {
+	constructor(sourceRequest, data, type, userRequest) {
+		super();
+		this.sourceRequest = sourceRequest;
+		this.request = data.id;
+		this.meta = data.meta;
+		this.type = type;
+		this.userRequest = userRequest;
+		this.built = false;
+		this.delegated = true;
```
Comment (sokra) [inline lib/DelegatedModule.js]: `delegated` is not in the original source.

Why adding it? If there is a reason, could you move it into a separate PR. The "refactor to ES6" PRs should only refactor to ES6...

Comment (davegonzalez) [inline lib/DelegatedModule.js]: https://github.com/webpack/webpack/blob/master/lib/DelegatedModule.js#L26

I was under the impression that 
`DelegatedModule.prototype.delegated = true;`

should be moved into the constructor. Does it belong elsewhere?

Comment (sokra) [inline lib/DelegatedModule.js]: ah ok. This was at a weird position. Thanks for fixing it.

**Comment by sokra** [comment]
Thanks

File: lib/DelegatedModule.js [hunk 2]
```diff
@@ -2,79 +2,83 @@
...
+		} else {
+			str = `module.exports = (__webpack_require__(${sourceModule.id}))`;
 
-DelegatedModule.prototype.unbuild = function() {
-	this.built = false;
-	Module.prototype.unbuild.call(this);
-};
+			switch(this.type) {
+				case "require":
+					str += `(${JSON.stringify(this.request)})`;
```
Comment (sokra) [inline lib/DelegatedModule.js]: Semicolon is missing here

File: lib/DelegatedModule.js [hunk 3]
```diff
@@ -2,79 +2,83 @@
...
+					break;
+				case "object":
+					str += `[${JSON.stringify(this.request)}]`;
```
Comment (sokra) [inline lib/DelegatedModule.js]: and here...

**Comment by sokra** [comment]
nevermind I do it myself.
